example/cajita_tutorial/10_spline/spline_example.cpp

```diff
@@ -0,0 +1,229 @@
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Cajita.hpp>
+#include <Kokkos_Core.hpp>
+
+#include <iostream>
+
+//---------------------------------------------------------------------------//
+// Spline example.
+//---------------------------------------------------------------------------//
+void splineExample()
+{
+    /*
+      Cajita provides B-Spline related interfaces for uniform grids operations.
+      The spline orders are:
+        0 - nearest grid point interpolation
+        1 - linear
+        2 - quadratic
+        3 - cubic
+    */
+    std::cout << "Cajita Spline Example\n" << std::endl;
+
+    /*
+      First, we need some setup to facilitate the use of Cajita splines.
+
+      This includes the creation of a simple uniform mesh and a single particle
+      position.
+    */
+    using ExecutionSpace = Kokkos::DefaultHostExecutionSpace;
+
+    // Create a 3-dimensional global uniform mesh
+    // Global bounding box.
+    double cell_size = 0.05;
+    std::array<int, 3> global_num_cell = { 20, 20, 20 };
+    std::array<double, 3> global_low_corner = { 0, 0, 0 };
+    std::array<double, 3> global_high_corner = {
+        global_low_corner[0] + cell_size * global_num_cell[0],
+        global_low_corner[1] + cell_size * global_num_cell[1],
+        global_low_corner[2] + cell_size * global_num_cell[2] };
+
+    // Create the global mesh
+    auto global_mesh = Cajita::createUniformGlobalMesh(
+        global_low_corner, global_high_corner, cell_size );
+
+    // Set up a uniform partitioner
+    Cajita::DimBlockPartitioner<3> partitioner;
+
+    // Create the global grid.
+    auto global_grid = Cajita::createGlobalGrid(
+        MPI_COMM_WORLD, global_mesh, { false, false, false }, partitioner );
+
+    // Get the local grid.
+    int halo_cell_width = 1;
+    auto local_grid = Cajita::createLocalGrid( global_grid, halo_cell_width );
+
+    // Get the local mesh for this rank.
+    auto local_mesh = Cajita::createLocalMesh<ExecutionSpace>( *local_grid );
+
+    // Create a particle position vector.
+    double xp[3] = { 0.5, 2, 1.5 };
+
+    /********************************************************************/
+    /*
+      There are two ways to use Cajita splines in practice.
+
+      In the first, one constructs an object of type:
+        SplineData<Scalar, Order, NumSpaceDim, EntityType, Tags...> where
+            Scalar is the underlying scalar type
+            Order = spline order
+            NumSpaceDim = number of spatial dimensions
+            EntityType = geometric entity type over which the spline is defined
+            Tags = variadic list of desired SplineDataMember tags.
```

Comment:
Might be worth having an example that shows using the tags. This is an expert user capability because all tags are on be default. Specifying tags can reduce memory/computation to setup the spline data but you need to know what your data dependencies are for your interpolation kernels (e.g. the distance as needed by APIC)

Reply:
Done.